code-challenges.test.js

```diff
@@ -11,46 +11,176 @@
+//   ✕ returns an array that length containing the numbers of the Fibonacci sequence using a number > than 2 (4 ms)
+
+// ● fibinochi › returns an array that length containing the numbers of the Fibonacci sequence using a number > than 2
+
+//   expect(received).toEqual(expected) // deep equality
+
+//   Expected: [1, 1, 2, 3, 5, 8]
+//   Received: undefined
+
+describe("fibinochi", () => {
+    it("returns an array that length containing the numbers of the Fibonacci sequence using a number > than 2", () => {
+      expect(fibinochi(fibLength1)).toEqual([1, 1, 2, 3, 5, 8]);
+     expect(fibinochi(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55]);
+    });
+  })
+
+ 
+
+
+// b) Create the function that makes the test pass.
+
+// ONLY WAS ABLE TO GET ON TEST TO PASS. SPENT THE WHOL WEEKEND ON JUST THIS
+
+
+let fibinochi= () => {}
+    const newArray = [0, 1];
+    for(var i = 2; i < newArray; i++) {
+       const a = (i - 1);
+       const b = (i - 1);
+       newArray.push(a + b);
+    }
+
+
+  
+       
+// QUESTION 2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+
+//PSUEDO CODING
+// create a functionn that takes in a an array
+// Output: new array of only odd numbers sorted from least to greatest.
+// Use the .filter method the filter through the array ang grab all of the odd numbers. Create a another variable for the new array, .sort through that array to get the number from leats to greatest
+// Expected output: -9, 7, 9, 199]
+// Expected output: 7,23, 67,-823]
+
+
 const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+// // Expected output: [-9, 7, 9, 199]
 
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
+// Expected output: [7,23, 67,-823]
 
 
-// b) Create the function that makes the test pass.
+// 1a) Create a test with expect statements for each of the variables provided.
 
+// FAIL  ./code-challenges.test.js
+// theOddNumbers
+//   ✕ returns a new array of only odd numbers sorted from least to greatest. (1 ms)
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// ● theOddNumbers › returns a new array of only odd numbers sorted from least to greatest.
+
+//   ReferenceError: theOddNumbers is not defined
+
+describe("theOdds", () => {
+    it("returns a new array of only odd numbers sorted from least to greatest.", () => {
+      expect(theOdds(fullArr1)).toEqual([-9, 7, 9, 199]);
+      expect(theOdds(fullArr2)).toEqual([7,23, 67,-823]);
+    });
+  })
+
+
+
+// 2b) Create the function that makes the test pass.
+
+// TRY One  🤷🏾‍♀️
+const onlyNumbers = fullArr1.filter( 
+    element => typeof element === "number"
+); 
+
+const odds = onlyNumbers.filter(number => {
+    return number % 2 !== 0;
+});
+
+const sorted= odds.sort
```

Comment:
A proper function should only be declared once and reused for each variable. Again check out whats going on with functions in javascript unit 1!